src/lib/datepicker/README.md

```diff
@@ -1,3 +1,181 @@
+then proceed to month view after choosing a month.
+
+The month or year that the calendar opens to is determined by first checking if any date is
+currently selected, if so it will open to the month or year containing that date. Otherwise it will
+open to the month or year containing today's date. This behavior can be overridden by using the
+`startAt` property of `md-datepicker`. In this case the calendar will open to the month or year
+containing the `startAt` date. 
+
+```html
+...
+<md-datepicker startView="year" startAt="1/1/1990"></md-datepicker>
+```
+
+### Preventing selection of specific dates
+There are two ways to restrict the dates available for selection in the datepicker. The first is by
+using the `min` and `max` properties of the input. This will disable all dates on the calendar
+before or after the respective given dates. It will also prevent the user from advancing the
+calendar past the `month` or `year` (depending on current view) containing the `min` or `max` date.
+ 
+The second way to restrict selection is using the `dateFilter` property of `md-datepicker`. The
+`dateFilter` property accepts a function of `<D> => boolean` (where `<D>` is the date type used by
+the datepicker, see section on
+[choosing a date implementation](#choosing-a-date-implementation-and-date-format-settings)).
+A result of `true` indicates that the date is selectable and a result of `false` indicates that it
+is not. One important difference between using `dateFilter` vs using `min` or `max` is that
+filtering out all dates before a certain point, will not prevent the user from advancing the
+calendar past that point.
+
+```ts
+myFilter = (d: Date) => d.getFullYear() > 2005 
+```
+
+```html
+<input [mdDatepicker]="d" min="1/1/2000" max="12/31/2020">
+<md-datepicker #d [dateFilter]="myFilter"></md-datepicker>
+```
+
+In this example the user can back past 2005, but all of the dates before then will be unselectable.
+They will not be able to go further back in the calendar than 2000.
+
+### Touch UI mode
+The datepicker normally opens as a popup under the input. However this is not ideal for touch
+devices that don't have as much screen real estate and need bigger click targets. For this reason
+`md-datepicker` has a `touchUi` property that can be set to `true` in order to enable a more touch
+friendly UI where the calendar opens in a large dialog.
+
+### Manually opening and closing the calendar
+The calendar popup can be programmatically controlled using the `open` and `close` methods on the
+`md-datepicker`. It also has an `opened` property that reflects the status of the popup.
+
+```ts
+@Component({...})
+export class MyComponent implements AfterViewInit {
+  @ViewChild(MdDatepicker) dp: MdDatepicker<Date>;
+  
+  ngAfterViewInit() {
+    dp.open();
+  }
+}
+```
+
+As seen in the example above, the `MdDatepicker` type is generic. This is because it can work with
```

Comment:
This should go into its own section rather than being under `### Manually opening and closing the calendar`. There should probably be one section for "Date formats and internationalization" (with sub-sections)